### DISCUSSION TITLE
MarkUp and session import seems messed up.

### DISCUSSION THREAD

**Discussion by Matthommes** [telegraf/telegraf#1945] [Q&A] [answered]
I'm trying to create a bot and I want to use Markup but I can't  seem to get the correct import.

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './markup' is not defined by "exports" in C:\Users\hp\Desktop\CheckBot\node_modules\telegraf\package.json
    at new NodeError (node:internal/errors:406:5)
    at exportsNotFound (node:internal/modules/esm/resolve:268:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:598:9)
    at resolveExports (node:internal/modules/cjs/loader:547:36)
    at Module._findPath (node:internal/modules/cjs/loader:621:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1034:27)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.<anonymous> (C:\Users\hp\Desktop\CheckBot\index.js:3:18) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}

Node.js v20.9.0

HERE's the error message

**Comment by MKRhere** [comment, marked as answer]
I can add this as a separate export in the upcoming release. The intended way to use it right now is

```ts
import { Markup } from "telegraf";
```

**Reply by Matthommes** [reply]
Thank you so much for your response. while that was very helpful, I still find it hard to use some of the other imports as i am following a tutorial that seem outdated.
file:///C:/Users/hp/Desktop/CheckBot/index.js:5
  Stage,
  ^^^^^
SyntaxError: Named export 'Stage' not found. The requested module 'telegraf' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'telegraf';

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:131:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:213:5)
    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:66:12)

Node.js v20.9.0
[nodemon] app crashed - waiting for file changes before starting...

this is the error i got trying to import and use Stage. I'm trying to build a bot that monitors and notifys me when i lose a follower in any of my social media platforms. any tips? what am i doing wrong???

**Reply by Matthommes** [reply]
import {
  Telegraf,
  Markup,
  session,
  Stage,
  WizardScene,
  Scenes,
} from "telegraf";
import dotenv from "dotenv";

dotenv.config();

// Import your scenes
import InstagramScene from "./scenes/InstagramScene.js";
import twitterScene from "./scenes/twitterScene.js";
import tiktokScene from "./scenes/tiktokScene.js";

const bot = new Telegraf(process.env.BOT_TOKEN);

// Create a new stage
const stage = new Stage();

// Register scenes
const instagramScene = new WizardScene("instagram-login", InstagramScene);
const twitterLoginScene = new WizardScene("twitter-login", twitterScene);
const tiktokLoginScene = new WizardScene("tiktok-login", tiktokScene);

stage.register(instagramScene);
stage.register(twitterLoginScene);
stage.register(tiktokLoginScene);

// Use the session and stage middleware
bot.use(session());
bot.use(stage.middleware());

bot.start(async (ctx) => {
  let name = ctx.from.first_name || ctx.from.username;
  let firstMessage = `Hey ${name}, welcome aboard!  Looking to stay on top of who follows and unfollows you across your social media?  Your personal follow/unfollow monitor is here to help! Connect your accounts to see instant notifications for new followers and unfollows, and stay ahead of the social game.  Ready to dive in?`;
  let secondMessage = "Which account would you want to signIn to?";
  let options = Markup.inlineKeyboard([
    Markup.button.callback("Twitter", "login_twitter"),
    Markup.button.callback("Instagram", "login_instagram"),
    Markup.button.callback("Tiktok", "login_tiktok"),
  ]);
  ctx.reply(firstMessage);
  ctx.reply(secondMessage, options);
});

bot.action("login_twitter", (ctx) => ctx.scene.enter("twitter-login"));

bot
  .launch()
  .then(() => {
    console.log("Bot started");
  })
  .catch((error) => {
    console.error("Error while starting bot", error);
  });

this is my current codebase.

**Reply by MKRhere** [reply]
Scenes.WizardScene might be the only change needed

**Reply by MKRhere** [reply]
And Scenes.Stage.
